control/controlclient/direct.go

```diff
@@ -262,6 +262,7 @@ func (c *Direct) doLogin(ctx context.Context, t *oauth2.Token, flags LoginFlags,
 	tryingNewKey := c.tryingNewKey
 	serverKey := c.serverKey
 	authKey := c.authKey
+	hostinfo := c.hostinfo
```

Comment:
This is a bit misleading. This is a pointer. What's guarded by c.mu, the pointer itself, or the stuff within it?

Below we access hostinfo.BackendLogID, so hopefully that can't change concurrently.

I think we need a bit more docs on ownership rules on the hostinfo field.

Reply:
This is justified by the fact that `c.hostinfo` is owned by `c` and never modified in-place. `PollNetMap` does the same thing for the same reason. This is very much non-obvious, so I'll elaborate on that.